generators/language-java/templates/java-spring/src/main/java/application/ibmcloud/CloudServices.java

```diff
@@ -141,8 +144,53 @@ private String getJsonValue(String jsonPath, String json) {
         }
         return value;
     }
-    
+  
     // Search pattern resolvers
+	
+    private String getUserProvidedValue(String pattern) {
+        LOGGER.info("user-provided entry found:  " + pattern);
+        String value = null;
+        String vcap_services = System.getenv(VCAP_SERVICES);
+        if (vcap_services == null || vcap_services.isEmpty() || pattern == null) {
+            LOGGER.info("No VCAP_SERVICES or no user-provided pattern");
+            return null;
+        }
+        int i = pattern.indexOf(":");
+        if (i == -1 || i == pattern.length() - 1) {
+            LOGGER.info("Invalid user-provided pattern");
+            return null;
+        }
+        String serviceName = pattern.substring(0, i);
+        String credentialKey = pattern.substring(i+1);
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            JsonNode vs = mapper.readTree(vcap_services);
+            ArrayNode array = (ArrayNode) vs.get("user-provided");
+            if (array.isArray()) {
+                LOGGER.info("Found user-provided array");
+                for (final JsonNode entryNode : array) {
+                    JsonNode nameNode = entryNode.get("name");
+                    LOGGER.info("Found user-provided array entry name field");
+                    if (nameNode != null) {
+                        LOGGER.info("user-provided array entry name: " + nameNode.textValue());
+                        String name = nameNode.textValue();
```

Comment:
I dont remember `textValue` I alwasy saw `asText()` for "getting" the value. I would double check the signature and hopefully there are some good unit test around this